improve: make sidecar work with readOnlyRootFileSystem

kolorful opened the pull request:
## Description

Add a emptyDir volume mount under `/tmp` on the agent sidecar so it works with `readOnlyRootFileSystem: true`

Current behavior with `readOnlyRootFileSystem` set to `true` on the pod.

```
error: mkdir /tmp/agent: read-only file system
```

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [x] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [x] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [x] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

kolorful committed:
improve: make sidecar work with readOnlyRootFileSystem

Signed-off-by: Kewei Ma <kewei@indeed.com>

kolorful commented:
I have tested and verified the functionality in our testing clusters.

kolorful commented:
@thallgren would you mind take a look this PR when you have time, I can't post in the slack cause I can't register with my email. Thanks!

thallgren commented:
It's generally a good idea to try the following before you submit a PR:
```
$ make check-unit
$ make lint             # this one is currently OK, but it's always good to check
```

kolorful commented:
> It's generally a good idea to try the following before you submit a PR:
> 
> ```
> $ make check-unit
> $ make lint             # this one is currently OK, but it's always good to check
> ```

Ah, good point. Updated and fixed the test.

kolorful commented:
Rebased over `release/v2` and pushed.
